odoo/addons/base/models/ir_asset.py

```diff
@@ -297,70 +342,48 @@ def _get_paths(self, path_def, installed, extensions=None):
             list of file paths matching the definition [1] (or the glob itself if
             none). Note that these paths are filtered on the given `extensions`.
         """
-        paths = []
-        path_url = fs2web(path_def)
-        path_parts = [part for part in path_url.split('/') if part]
+        paths = None
+        path_parts = [part for part in path_def.split(os.sep) if part]
         addon = path_parts[0]
         addon_manifest = odoo.modules.module.get_manifest(addon)
 
         safe_path = True
         if addon_manifest:
             if addon not in installed:
                 # Assert that the path is in the installed addons
-                raise Exception("Unallowed to fetch files from addon %s" % addon)
-            addons_path = os.path.join(addon_manifest['addons_path'], '')[:-1]
-            full_path = os.path.normpath(os.path.join(addons_path, *path_parts))
-
-            # first security layer: forbid escape from the current addon
-            # "/mymodule/../myothermodule" is forbidden
-            # the condition after the or is to further guarantee that we won't access
-            # a directory that happens to be named like an addon (web....)
-            if addon not in full_path or addons_path not in full_path:
-                addon = None
-                safe_path = False
-            else:
+                raise Exception(f"Unallowed to fetch files from addon {addon} for file {path_def}")
+            addons_path = addon_manifest['addons_path']
```

Comment:
As far as i understand the previous version was trying to remove a trailling /. It looks like the addons_path won't have a trailling / and that this check is not useful. In any case would be easier to make it conditional, with endswith() instead of an expensive path.join